app/models/agents/delay_agent.rb

```diff
@@ -11,13 +11,16 @@ class DelayAgent < Agent
 
       `expected_receive_period_in_days` is used to determine if the Agent is working. Set it to the maximum number of days
       that you anticipate passing without this Agent receiving an incoming Event.
+
+      `max_emitted_events` is used to limit the number of the maximum events which should be created. If you omit this DelayAgent will create events for every event stored in the memory.
     MD
 
     def default_options
       {
         'expected_receive_period_in_days' => "10",
         'max_events' => "100",
-        'keep' => 'newest'
+        'keep' => 'newest',
+        'max_emitted_events' => "100"
```

Comment:
Maybe we shouldn't include this in the default options so that everything is emitted by default?